Software/Frontend/app/src/main/java/hr/foi/air/baufind/ui/screens/JobSearchScreen/JobSearchScreen.kt

```diff
@@ -31,14 +31,18 @@ import hr.foi.air.baufind.ui.components.PrimaryTextField
 import hr.foi.air.baufind.ws.network.TokenProvider
 import kotlinx.coroutines.launch
 
-//moram za oba errora prikazat tako gdje bi bila lista poruku pogreške
-
 @Composable
 fun JobSearchScreen(navController: NavController, tokenProvider: TokenProvider){
     val coroutineScope = rememberCoroutineScope()
     var jobSearchResponse by remember { mutableStateOf<JobSearchResponse?>(null) }
 
     var searchText by remember { mutableStateOf("") }
+    val filteredJobs = remember(jobSearchResponse, searchText){
+        jobSearchResponse?.jobs?.filter { job ->
+            job.title.contains(searchText, ignoreCase = true) ||
+                    job.skills.any { skill -> skill.title.contains(searchText, ignoreCase = true)}
+        } ?: emptyList()
+    }
```

Comment:
Sad ne znam je li ovo i na drugim mjestima ovako, ali malo je možda zbunjujuće što zovemo "Response", a ne kao "ViewModel"? Što predlažete da onda radimo? Jer ovako miješamo i "Response" kao data klase za prikaz i "Response" klase za odgovore od API-ja.

Reply:
Ok, mogu napraviti neki JobSearchViewModel koji će u sebi imati metodu da loada podatke tako što poziva API jer će mu se dati tokenProvider

I onda sa povratnim podatcima od API-a napuni svoj atribut (ovdje npr jobs) i onda se taj ViewModel korist iza rad s podatcima. 

Tako onda ne moramo direktno pozivat service na ekranima nego u viewmodelima, a i imamo konzistentniji nacin pristupanju podatcima

Reply:
Da, to sa više response je malo čudno, definitivno trebamo obratiti pozornost da to sredimo iduci sprint.
To sa viewModel je dobra stvar, koristio sam isto to za profil. Tako da vidi probaj.